File: dataModule/src/main/java/com/mobnetic/coinguardian/model/FuturesMarket.java
package com.mobnetic.coinguardian.model;

import java.util.HashMap;

public abstract class FuturesMarket extends Market {
	
	public final int[] contractTypes;
			
	public FuturesMarket(String name, String ttsName, HashMap<String, CharSequence[]> currencyPairs, int[] contractTypes) {
		super(name, ttsName, currencyPairs);
		this.contractTypes = contractTypes;
	}

	@Override
	public final String getUrl(int requestId, CheckerInfo checkerInfo) {
		return getUrl(requestId, checkerInfo, checkerInfo.getContractType());
	}
	
	protected abstract String getUrl(int requestId, CheckerInfo checkerInfo, int contractType);
	
}

File: dataModule/src/main/java/com/aneonex/bitcoinchecker/datamodule/model/FuturesMarket.kt
package com.aneonex.bitcoinchecker.datamodule.model

import com.aneonex.bitcoinchecker.datamodule.model.currency.CurrencyPairsMap

abstract class FuturesMarket(name: String, ttsName: String, currencyPairs: CurrencyPairsMap?, val contractTypes: IntArray)
    : Market(name, ttsName, currencyPairs) {

    override fun getUrl(requestId: Int, checkerInfo: CheckerInfo): String {
        return getUrl(requestId, checkerInfo, checkerInfo.contractType)
    }

    protected abstract fun getUrl(requestId: Int, checkerInfo: CheckerInfo, contractType: Int): String
}